cdap-data-fabric/src/main/java/io/cdap/cdap/spi/metadata/DelegatingMetadataStorage.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright © 2025 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.spi.metadata;
+
+import com.google.inject.Inject;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Delegates {@link MetadataStorage} based on configured extension.
+ */
+public class DelegatingMetadataStorage implements MetadataStorage {
+  private final CConfiguration cConf;
+  private final MetadataStorage delegate;
+  private static String prefix;
+
+  @Inject
+  DelegatingMetadataStorage(CConfiguration cConf, MetadataStorageExtensionLoader extensionLoader) throws Exception {
+    this.cConf = cConf;
+    this.delegate = extensionLoader.get(getName());
+
+    if (this.delegate == null) {
+      throw new IllegalArgumentException("Unsupported MetadataProvider type: " + getName());
+    }
+    // TODO(CDAP-21174): Create metadata storage specific properties
+    if (getName().equals("gcp-spanner")) {
+      this.prefix = String.format("%s%s.", Constants.Dataset.STORAGE_EXTENSION_PROPERTY_PREFIX, getName());
+    }
+    else{
```

Comment:
add space after `else`

Can you please use `Google Configuration Checker Style`? It will automatically take care of these issues.